File: frontend/src/types/charge-library.ts
export interface ChargeLibraryItem {
  id: string;
  charge_key: string;
  charge_name: string;
  category: 'delivery_transportation' | 'services' | 'labor' | 'other';
  description: string | null;
  is_enabled: boolean;
  is_system: boolean;
  pricing_type: 'fixed' | 'variable' | 'per_mile' | 'tiered';
  fixed_amount: number | null;
  per_mile_rate: number | null;
  free_radius_miles: number | null;
  zone_config: ZoneConfig[] | null;
  guidance_min: number | null;
  guidance_max: number | null;
  variable_placeholder: string | null;
  auto_suggest: boolean;
  auto_suggest_trigger: string | null;
  invoice_label: string | null;
  sort_order: number;
  notes: string | null;
}

export interface ZoneConfig {
  zone_number: number;
  max_miles: number | null; // null = "beyond previous zone"
  price: number;
}

export interface ChargeUpdate {
  charge_key: string;
  is_enabled: boolean;
  pricing_type: string;
  fixed_amount: number | null;
  per_mile_rate: number | null;
  free_radius_miles: number | null;
  zone_config: ZoneConfig[] | null;
  guidance_min: number | null;
  guidance_max: number | null;
  variable_placeholder: string | null;
  auto_suggest: boolean;
  auto_suggest_trigger: string | null;
  invoice_label: string | null;
  notes: string | null;
}

export const CATEGORY_LABELS: Record<string, string> = {
  delivery_transportation: 'Delivery & Transportation',
  services: 'Services',
  labor: 'Labor',
  other: 'Other',
};

export const CATEGORY_COLORS: Record<string, string> = {
  delivery_transportation: 'bg-blue-100 text-blue-700',
  services: 'bg-purple-100 text-purple-700',
  labor: 'bg-amber-100 text-amber-700',
  other: 'bg-gray-100 text-gray-700',
};

export const TRIGGER_DESCRIPTIONS: Record<string, string> = {
  always: 'Suggested on every order',
  after_hours: 'Suggested when delivery time is outside 7am-5pm',
  rush_48h: 'Suggested when delivery is within 48 hours of order date',
  weekend: 'Suggested on Saturday and Sunday deliveries',
};

File: backend/app/schemas/charge_library.py
"""Charge Library schemas — request/response models for fees and surcharges."""

from pydantic import BaseModel, ConfigDict


class ChargeLibraryItemResponse(BaseModel):
    id: str
    charge_key: str
    charge_name: str
    category: str
    description: str | None
    is_enabled: bool
    is_system: bool
    pricing_type: str
    fixed_amount: float | None
    per_mile_rate: float | None
    free_radius_miles: float | None
    zone_config: list[dict] | None
    guidance_min: float | None
    guidance_max: float | None
    variable_placeholder: str | None
    auto_suggest: bool
    auto_suggest_trigger: str | None
    invoice_label: str | None
    sort_order: int
    notes: str | None

    model_config = ConfigDict(from_attributes=True)


class ChargeLibraryItemUpdate(BaseModel):
    charge_key: str
    is_enabled: bool = False
    pricing_type: str = "variable"
    fixed_amount: float | None = None
    per_mile_rate: float | None = None
    free_radius_miles: float | None = None
    zone_config: list[dict] | None = None
    guidance_min: float | None = None
    guidance_max: float | None = None
    variable_placeholder: str | None = None
    auto_suggest: bool = False
    auto_suggest_trigger: str | None = None
    invoice_label: str | None = None
    notes: str | None = None


class ChargeLibraryBulkSaveRequest(BaseModel):
    charges: list[ChargeLibraryItemUpdate]


class ChargeLibraryItemCreate(BaseModel):
    charge_name: str
    category: str = "other"
    description: str | None = None
    pricing_type: str = "variable"
    fixed_amount: float | None = None
    invoice_label: str | None = None
